src/routes/FriendList.tsx

```diff
@@ -0,0 +1,52 @@
+import styled from 'styled-components';
+import user from '../data/user.json';
+import Friends from '../components/FriendList/Friends';
+import UnderNavBar from '../components/layout/UnderNavBar';
+import { HeaderContains, HeaderText } from '../components/layout/CommonStyle';
+import SearchUser from '../components/FriendList/SearchUser';
+import { useState } from 'react';
+import { AiOutlineUser } from 'react-icons/ai';
+
+const FriendList = () => {
+  const [searchedUser, setsearchedUser] = useState(user);
+  const filteredUser = (searchText: string) => {
+    if (!searchText) {
+      setsearchedUser(user);
+    } else {
+      const filtered = user.filter((element) => {
+        return element.userName
+          .toUpperCase()
+          .includes(searchText.toUpperCase());
+      });
+      setsearchedUser(filtered);
+    }
+  };
```

Comment:
검색할 때 `setsearchedUser(filtered)` 부분에서 무한루프가 발생합니다. 이는 렌더링되는 과정에서 state가 변화되어 리렌더링이 반복적으로 일어나 결국 무한루프에 빠져 버리는 현상인데요, 아래와같이 `useCallback`을 사용하시면 이를 해결할 수 있습니다!

```suggestion
  const filteredUser = (searchText: string) => {
    if (!searchText) {
      setsearchedUser(user);
    } else {
      const filtered = user.filter((element) => {
        return element.userName
          .toUpperCase()
          .includes(searchText.toUpperCase());
      });
      setsearchedUser(filtered);
    }
  };
```

Reply:
useInput hook에 useCallback을 추가해봤는데 이렇게 수정해도 무한루프 현상이 해결되는게 맞을까요,,,?